ledger/src/shred.rs

```diff
@@ -799,6 +805,11 @@ where
         stats.index_bad_deserialize += 1;
         return true;
     };
+    let Some(fec_set_index) = layout::get_fec_set_index(shred) else {
+        stats.fec_set_index_bad_deserialize += 1;
```

Comment:
these deserialize failures are already thrown out when we parse the shred in sigverify, this just moves it up - shouldn't be a consensus altering change no need to feature flag.

Reply:
sounds reasonable